clang/lib/Sema/SPIRVBuiltins.td

```diff
@@ -952,3 +952,7 @@ foreach name = ["GroupUMin", "GroupUMax"] in {
 foreach name = ["GroupSMin", "GroupSMax"] in {
   def : SPVBuiltin<name, [ASIGenTypeN, UInt, UInt, ASIGenTypeN], Attr.Convergent>;
 }
+
+foreach name = ["GroupLogicalAndKHR", "GroupLogicalOrKHR"] in {
+  def : SPVBuiltin<name, [Bool, UInt, UInt, Bool], Attr.Convergent>;
```

Comment:
According to [`SPV_KHR_uniform_group_instructions`](https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/KHR/SPV_KHR_uniform_group_instructions.asciidoc):

> 'Result Type' must be a scalar or vector of [Boolean type](https://github.com/KhronosGroup/SPIRV-Registry/blob/main/extensions/KHR/SPV_KHR_uniform_group_instructions.asciidoc#Boolean%20type).
> The type of 'X' must be the same as 'Result Type'.

I guess this change misses boolean vectors support. Note: I'm fine if that is outlined into a separate PR, but then we probably need to update the title to explicitly mention that only scalars are supported for now.

Reply:
Good catch, on the SYCL side, I think vector types should still work with the logical operators group algorithms, as the the current implementation of the group algorithms implement the vector group algorithms using the scalar group algorithms. But the SPIRV builtin will only be able to support a scalar value, so I can add a TODO comment mentioning that. I'll also add a test that uses vector types.

Reply:
There are a couple issues with the current implementation for group algorithms for vector types and the return type of `operator&&` for `vec<bool, N>`, so I will address those issues in another PR. I've renamed the PR to only mention support for scalar operations now.

Reply:
> There are a couple issues with the current implementation for group algorithms for vector types and the return type of `operator&&` for `vec<bool, N>`, so I will address those issues in another PR. I've renamed the PR to only mention support for scalar operations now.

Works for me, thanks!